nomos-services/mempool/src/lib.rs

```diff
@@ -195,6 +203,9 @@ where
         loop {
             tokio::select! {
                 Some(msg) = service_state.inbound_relay.recv() => {
+                    if let Some(metrics) = &self.metrics {
+                        metrics.record(&msg);
+                    }
```

Comment:
I think all metrics stuff on services should be feature gated as well 🤔

Reply:
I can feature gate it, everything could be contained inside the `metrics.rs` in each service, but I don't thing that it's necessary - just like in our previous approach, we didn't feature gate the individual messages, here we don't record anything and disable the service responsible for gathering what's collected.

Reply:
Yeah, I understand it. I just have an itch on having a `if` check for something it may never be true running on production. Let's see if someone else back up or we just leave it like this.

Reply:
Now I get it. I was betting on cpu branch prediction, but rust could just eliminate the problem, will add a feature gate.